Features/Assets/Sources/ViewModels/AssetSceneViewModel.swift

```diff
@@ -178,6 +178,18 @@ public final class AssetSceneViewModel: Sendable {
     var priceAlertsViewModel: PriceAlertsViewModel {
         PriceAlertsViewModel(priceAlerts: assetData.priceAlerts)
     }
+    
+    var swapAssetType: SelectedAssetType {
+        switch assetData.asset.type {
+        case .native: .swap(assetData.asset, nil)
+        case .erc20, .bep20, .spl, .spl2022, .trc20, .token, .ibc, .jetton, .synth, .asa, .perpetual, .spot:
```

Comment:
```suggestion
        case .token:
```

there is a `subtype`